sova/core/steps/spec.py

```
@@ -216,15 +231,41 @@ def _text_has_open_questions(text: str) -> bool:
+    context_files = [
+        (project_dir / ".claude" / "rules" / "architecture.md", "Architecture"),
+        (project_dir / "AGENTS.md", "Project Conventions"),
+    ]
+    for path, label in context_files:
+        if path.is_file():
+            try:
+                content = path.read_text()
+                if len(content) > 8000:
+                    content = content[:8000] + "\n... (truncated)"
+                sections.append(f"### {label}\n\n{content}")
+            except IOError:
```

Ruff (0.16.1)

[warning] Replace aliased errors with `OSError`

Replace `IOError` with builtin `OSError`

(UP024)